feature/gnmi/telemetry_npu_asic_counters_test/tests/README.md

```diff
@@ -0,0 +1,176 @@
+
+*   Send IPinIP traffic to OuterDstIP_1. Validate that ATE port-2 receives the IPinIP traffic.
+
+*   Packet Counters - Interface-Block
+
+    *   Subscribe to the following, and capture values of the 4 leaves (1).
+        * /packet/interface-block/in-pkts
+        * /packet/interface-block/out-pkts
+        * /packet/interface-block/in-bytes
+        * /packet/interfac-block/out-bytes
+    *   Wait for 30 secs and capture the values of 4 leaves(2). 
+    *   Calculation of step 2 - step 1 leaf values. 
+        * 30*pps = in-pkts and out-pkts
+        * 30*pps*300 = in-bytes and out-bytes
+    * Stop traffic.
+
+*   Packet Counters - Host-Block
+
+    *   Subscribe to the following, and capture the values of the 4 leaves(1),
+        * /packet/host-interface-block/in-pkts
+        * /packet/host-interface-block/out-pkts
+        * /packet/host-interface-block/in-bytes (TBD)
+        * /packet/host-interface-block/in-bytes (TBD)
+    *   Define variance = 0.99
+    *   Send ping packets to ATEPort2IP with count 1000. Wait for 20 secs.
+    *   Capture the in-pkts and out-pkts leaves(2).
+    *   Calculation of step 2 - step 1 leaf values.
+        * Step 2 - Step 1 >= 1000
+        * Step 2/Step 1 >= variance
+    *   Verify in-bytes and out-bytes (TBD)
+
+*   Packet Countes - Queueing Block
+
+    *   Subscribe to the following, and capture the values of the 4 leaves(1),
+        * /packet/queueing-block/in-pkts(TBD)
+        * /packet/queueing-block/in-bytes(TBD)
+        * /packet/queueing-block/out-pkts,
+        * /packet/queueing-block/out-bytes
+    *   Define variance = 0.98.
+    *   Start traffic.
+    *   Wait for 30 secs.
+    *   Capture the out-pkts and out-bytes leaf values (2).
+    *   Calculation of step 2 - step 1.
+        *  30 * pps = out-pkts.
+        *  30 * pps * 300 = out-bytes.
+        *  Capture tgen rx packets, tgen rx packets/step(2) values >= variance.
+    *   Stop traffic.
+
+
+*   Drop Counters - Lookup-Block
+
+    *   Subscribe to the following, and capture the leaf values(1)
+        * /drop/lookup-block/no-route
+        * /drop/lookup-block/no-nexthop
+        * /drop/lookup-block/invalid-packet
+        * /drop/lookup-block/forwarding-policy (TBD)
+        * /drop/lookup-block/incorrect-software-state
+        * /drop/lookup-block/rate-limit (TBD)
+        * /drop/lookup-block/acl-drops
+    
+    *   Start traffic, verify traffic received on atePort2.
+    *   Shutdown ateport2. Wait for 30 secs.
+    *   Verify /drop/lookup-block/no-route counter populated (2).
+        * Step 2-Step 1(no-route), calculation 30 * pps.       
+    *   Unshut ateport2, verify traffic recovers on atePort2.
+
+    *   Configure static Null0 route for outerDst_IP1.
+    *   After 30 secs, Verify /drop/lookup-block/no-nexthop leaf populated(3).
+        * Step 3- Step 1(no-nexthop), calcuation 30 * pps.
+    *   Unconfigure static route, and verify traffic recovers on atePort2, stop traffic.
```

Comment:
I'm not sure if Null0 is "no-nexthop". Actually Null0 is a next-hop.
The better test would be:
- remove NH#1 by gRIBI w/o removing NGH#1.

Reply:
No-nexhop can also simulated with glean adjacency.